augur/io/metadata.py

```diff
@@ -424,3 +426,47 @@ def write_records_to_tsv(records, output_file):
 
         for record in records:
             tsv_writer.writerow(record)
+
+
+class Metadata(TabularFile):
+    def __init__(self, file:str,
+            header:Optional[bool]=True,
+            names:Optional[List[str]]=None,
+            potential_id_columns:Optional[List[str]]=None):
+        super().__init__(file, header, names)
+
+        self.potential_id_columns = potential_id_columns or ("strain", "name")
+        """Metadata columns that may contain identifier information."""
```

Comment:
From [PEP 257 – Docstring Conventions](https://peps.python.org/pep-0257/#what-is-a-docstring):

> String literals occurring immediately after a simple assignment at the top level of a module, class, or `__init__` method are called “attribute docstrings”.

These are used by VSCode to show the docstring upon mouse hover elsewhere in the code. I suspect other editors that use docstrings to enrich the experience might also support them.

![image](https://user-images.githubusercontent.com/13424970/206322457-7298ac44-c9ab-4ffa-ab62-f460b8f98d9a.png)

Reply:
sphinx.ext.autodoc also supports this kind of less-frequently-seen docstring.

Reply:
Got it. It feels a bit like catering to a computer's ability to read the code rather than a human's, but if the dev experience for most people benefits from this approach it makes sense. I would not generally want these types of comments in sphinx docs, though...

Reply:
I think @huddlej may have been referring to other more verbose comments that are used around the codebase, which are only useful when you're trying to understand the underlying logic. The numpy-style docstrings also contain descriptions that are equivalent to the docstrings I'm using here, which are already rendered on the Developer API doc pages.

Reply:
@tsibley I definitely want class and instance attributes to be documented. My initial concern was that other variable assignments that use attribute docstrings would get pulled into API docs, too, but [PEP 258](https://peps.python.org/pep-0258/#attribute-docstrings) is clearer that this this shouldn't happen.

I was also expecting class and instance attributes to be documented by class- and init-level docstrings based on the more traditional guidance from PEP 257:

> The docstring for a class should summarize its behavior and list the public methods and instance variables.

It's cool that the attribute docstrings appear in VSCode and other docutils processing systems, but they don't show up on the Python or ipython REPLs where I would look for them during usage and debugging. For example, if I define `AClass` from the [PEP 258 example code](https://peps.python.org/pep-0258/#attribute-docstrings), I can't access the attribute docstrings with `AClass.c.__doc__`. A class- or init-level docstring would be accessible from the REPLs and provide the same information via `AClass.__doc__` or `AClass().__doc__`. For example, the following version of `AClass` provides information about class attributes on the REPL and API docs:

```python
class AClass:
    """A class!

    Attributes
    ----------
    c : str
       This is AClass.c's docstring. 
    i : str
        This is self.i's docstring.

    """
    c = 'class attribute'

    def __init__(self):
        """Method __init__'s docstring."""

        self.i = 'instance attribute'
```

From the REPL, I can get the docstring with:

```python
>>> print(AClass.__doc__)
A class!

    Attributes
    ----------
    c : str
       This is AClass.c's docstring.
    i : str
        This is self.i's docstring.


>>>
```